src/bootstrap/client-bootstrapper.js

```diff
@@ -46,6 +46,7 @@ export async function hydrateApp({ bundleConfig }) {
     pageName: window.__pageName__,
     container,
     initialState: window.__initalState__,
+    theme: window.__theme__,
```

Comment:
We should also, somewhere above in this file, register the `ThemeManager` with the container (like with the `ResourceManager` above), so it has the correct values set when it is requested from the container (I know it currently isn't in client-running code yet, but it could):

~~~
const themeManager = new ThemeManager(window.__theme__);
container.registerInstance(ThemeManager, themeManager);
~~~

Reply:
Yes of course, registered.

Reply:
Sorry, I hadn't seen the edited version of this comment.

Reply:
Maybe I didn't get it right. The `ThemeManager` constructor expects an object with the less variables and constructs a theme object out of them. But on the client, in `window.__theme__` we already have the constructed theme. In the code you don't seem to distinguish :thinking:

If I get the instance on the browser like this:

~~~
  const themeManager = useService(ThemeManager);
  console.log(themeManager.getTheme());
~~~

... I get:

~~~
number-utils.js:2 Uncaught TypeError: Cannot read properties of undefined (reading 'match')
    at getNumberFromString (number-utils.js:2:26)
~~~

Reply:
Another deadlock :laughing:

Reply:
Oh yes, I see, that was completely wrong what I had done there actually. Corrected it, thanks!

Reply:
:+1: